src/main/java/ratpack/zipkin/internal/DefaultServerTracingHandler.java

```diff
@@ -33,16 +33,13 @@ public DefaultServerTracingHandler(HttpTracing httpTracing) {
   public void handle(Context ctx) throws Exception {
     Request request = ctx.getRequest();
     final Span span = handler.handleReceive(extractor, request);
-
+    final Tracer.SpanInScope scope = tracing.tracer().withSpanInScope(span);
     ctx.getResponse().beforeSend(response -> {
       handler.handleSend(response, null, span);
-      span.finish();
+      scope.close();
     });
-
-    try(Tracer.SpanInScope ws = tracing.tracer().withSpanInScope(span)) {
-      span.start();
-      ctx.next();
-    }
+    span.start();
```

Comment:
As I mentioned above, think this `try-with-resources` block actually completes almost immediately.